test/runtests.jl

```diff
@@ -215,4 +215,31 @@ end
             @test sort(x; rev=true) == sort(y; rev=true)
         end
     end
+end
+
+@testset "inlinestrings" begin
+
+    @test inlinestrings([]) == Union{}[]
+    
+    x = inlinestrings("$i" for i in (1, 10, 100))
+    @test eltype(x) === String3
+    @test x == ["1", "10", "100"]
+    @test eltype(inlinestrings([randstring(i) for i = 1:3])) === String3
+    @test eltype(inlinestrings([randstring(i) for i = 1:4])) === String7
+    @test eltype(inlinestrings(["a", missing, "abcd"])) === Union{String7, Missing}
+    @test eltype(inlinestrings([missing, "abcd"])) === Union{String7, Missing}
+
+    # Base.SizeUnknown() case
+    t() = true
+    x = inlinestrings("$i" for i in (1, 10, 100) if t())
+    @test eltype(x) === String3
+    @test x == ["1", "10", "100"]
+    @test eltype(inlinestrings(randstring(i) for i = 1:3 if t())) === String3
+    @test eltype(inlinestrings(randstring(i) for i = 1:4 if t())) === String7
+    @test eltype(inlinestrings(x for x in ["a", missing, "abcd"] if t())) === Union{String7, Missing}
+    @test eltype(inlinestrings(x for x in [missing, "abcd"] if t())) === Union{String7, Missing}
+
+    x = [randstring(i) for i = 1:31]
+    @test InlineString.(x) == map(InlineString, x) == collect(InlineString, x)
```

Comment:
should we also test that these  dispatch to `inlinestrings` and so elements have a homogenous type?
```suggestion
    @test InlineString.(x) == map(InlineString, x) == collect(InlineString, x)
    @test eltype(InlineString.(x)) == eltype(map(InlineString, x)) == eltype(collect(InlineString, x)) == InlineString31
```